### PR TITLE
FIX Disable the check of None user_id in get_visible_bed_counts_for_user

### PR DESCRIPTION
**Pull request by rth**
Re-opens https://github.com/icubam/icubam/issues/180

The fix in https://github.com/icubam/icubam/issues/180 was not sufficient as it also needs to account for `force=True`. For now just revert it.

### CODE REVIEW

**Comment by dulacarnold** [comment]
We need integration tests, this only triggered because I checked by hand on the staging server...  Actually we need: TornadoTests on the API, that would have found it, and also some automated tests on staging...  But good fix, I was digging through trying to figure out when this broke :)

**Comment by maximedenes** [comment]
At least the call to `user.is_admin` should be guarded.

**Comment by rth** [comment]
Good thing it wasn't in production. I agree we need better tests.

> At least the call to user.is_admin should be guarded.

Yes, let me do something a bit better that would also keep mypy happy.

**Comment by maximedenes** [comment]
What's the spec of `force`?

**Comment by maximedenes** [comment]
Probably we should:
- guard the `user.is_admin` call
- make sure the user is real in the `else` branch
- add a brief description of the spec of `force` as a comment

**Comment by dulacarnold** [comment]
> What's the spec of `force`?

@olivierteboul can comment better, but force was basically there it to allow us to continue operating while proper authentication was put in place.  Because the API handles authentication differently (no associated user), force is the only way currently to get the store to provide necessary data, and allow the API to manage auth through key-based mechanisms.

File: icubam/db/store.py [hunk 1]
```diff
@@ -755,11 +755,17 @@ def get_visible_bed_counts_for_user(
...
     user = self.get_user(user_id)
-    if not user:
-      raise ValueError(f"Cannot find user with id {user_id}")
-    if force or user.is_admin:
+    if force:
+      return self._get_bed_counts_for_icus(None, latest=True, **kargs)
+    elif user is None:
+      raise ValueError("a non null user_id must be provided")
+    elif user.is_admin:
       return self._get_bed_counts_for_icus(None, latest=True, **kargs)
```
Comment (rth) [inline icubam/db/store.py:769]: OK I refactored a bit. Now it's a bit verbose (and there is one repetition) but mypy is able to correctly narrow down the types of `user` from `Optional[int]` to `int` after the None check.

Previously with,
```
if user is not None and not force:
    raise ValueError
```
the type after the check could still be None.

File: icubam/db/store.py [hunk 2]
```diff
@@ -755,11 +755,17 @@ def get_visible_bed_counts_for_user(
...
+      force: run query as admin irrespective of the user_id
     """
     user = self.get_user(user_id)
-    if not user:
-      raise ValueError(f"Cannot find user with id {user_id}")
-    if force or user.is_admin:
+    if force:
+      return self._get_bed_counts_for_icus(None, latest=True, **kargs)
+    elif user is None:
+      raise ValueError("a non null user_id must be provided")
```
Comment (dulacarnold) [inline icubam/db/store.py:767]: Maybe use capitals and punctuation while we're at it.

Comment (maximedenes) [reply icubam/db/store.py:767]: Also, maybe `user_id` is not actually null be just the user can't be found, right?

File: icubam/db/store.py [hunk 3]
```diff
@@ -755,11 +755,17 @@ def get_visible_bed_counts_for_user(
...
+      user_id: run query for a given user
+      force: run query as admin irrespective of the user_id
     """
     user = self.get_user(user_id)
-    if not user:
-      raise ValueError(f"Cannot find user with id {user_id}")
-    if force or user.is_admin:
+    if force:
+      return self._get_bed_counts_for_icus(None, latest=True, **kargs)
+    elif user is None:
```
Comment (dulacarnold) [inline icubam/db/store.py:766]: Could we just do `elif user is None and not force:`?

File: icubam/db/store.py [hunk 4]
```diff
@@ -755,11 +755,17 @@ def get_visible_bed_counts_for_user(
...
+      raise ValueError("a non null user_id must be provided")
+    elif user.is_admin:
```
Comment (dulacarnold) [inline icubam/db/store.py:768]:
```suggestion
    elif user.is_admin or force:
```

File: icubam/db/store.py (hunk 2 above)
Comment (dulacarnold) [inline icubam/db/store.py]:
```suggestion
      raise ValueError("A non null user_id must be provided.")
```

File: icubam/db/store.py (hunk 4 above)
Comment (maximedenes) [reply icubam/db/store.py:768]: Don't we already know `force == false` at that point?

**Pull request merged**